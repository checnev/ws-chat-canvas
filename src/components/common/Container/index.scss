$breakPoints: (xs: 440, sm: 600, md: 768, lg: 992, xl: 1280, xxl: 1920);

.container {
  width: 100%;
  display: flex;
  flex-wrap: wrap;
  padding-left: 15px;
  padding-right: 15px;
  margin: auto;

  @each $point, $maxWidth in $breakPoints {
    &_maxWidth_#{$point} {
      max-width: #{$maxWidth}px;
    }
  }

  &_flexDirection {
    &_column {
      flex-direction: column;
    }

    &_row {
      font-display: row;
    }
  }

  &_justifyContent {
    &_flex-start {
      justify-content: flex-start;
    }

    &_flex-end {
      justify-content: flex-end;
    }

    &_center {
      justify-content: center;
    }

    &_space-around {
      justify-content: space-around;
    }

    &_space-between {
      justify-content: space-between;
    }
  }

  &_alignItems {
    &_flex-start {
      align-items: flex-start;
    }

    &_flex-end {
      align-items: flex-end;
    }

    &_center {
      align-items: center;
    }
  }

  &_flexWrap {
    &_wrap {
      flex-wrap: wrap;
    }

    &_nowrap {
      flex-wrap: nowrap;
    }
  }
  
}
