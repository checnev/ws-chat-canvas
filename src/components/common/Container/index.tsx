import React from 'react';
import cn from 'utils/cn';
import './index.scss';

export type JustifyContent = 'flex-start' | 'center' | 'flex-end' | 'space-around' | 'space-between';
export type AlignItems = 'flex-start' | 'center' | 'flex-end';
export type FlexWrap = 'nowrap' | 'wrap';
export type FlexDirection = 'row' | 'column';
export type BreakPoint = 'xs' | 'sm' | 'md' | 'lg' | 'xl' | 'xxl';

interface ContainerProps {
  justifyContent?: JustifyContent;
  alignItems?: AlignItems;
  flexWrap?: FlexWrap;
  flexDirection?: FlexDirection;
  maxWidth?: BreakPoint;
}

class Container extends React.Component<ContainerProps> {
  render() {
    const container = cn('container');
    const { children, ...styleProps } = this.props;

    return (
      <div
        className={container(styleProps as Record<string, string>)}
      >
        {children}
      </div>
    );
  }
}

export default Container;
