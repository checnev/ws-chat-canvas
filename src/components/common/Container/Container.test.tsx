import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import Container from 'components/common/Container';

let container: any = null;

beforeEach(() => {
  container = document.createElement('div');
  document.body.append(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('Container component', () => {
  const setUp = (props?: any) => (<Container {...props}>Container Text</Container>);

  it('should render Container', () => {
    act(() => {
      render(setUp(), container);
    });

    expect(
      pretty(container.innerHTML)
    ).toMatchSnapshot();
  });

  it('should render customized Container', () => {
    const props = {
      maxWidth: 'lg',
      justifyContent: 'center',
      alignItems: 'center',
      flexWrap: 'wrap',
      FlexDirection: 'row',
    };

    act(() => {
      render(setUp(props), container);
    });

    expect(
      pretty(container.innerHTML)
    ).toMatchSnapshot();
  });
});
