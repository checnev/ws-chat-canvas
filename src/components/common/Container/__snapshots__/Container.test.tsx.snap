// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component should render Container 1`] = `"<div class=\\"container\\">Container Text</div>"`;

exports[`Container component should render customized Container 1`] = `"<div class=\\"container container_maxWidth_lg container_justifyContent_center container_alignItems_center container_flexWrap_wrap container_FlexDirection_row\\">Container Text</div>"`;
